### PR TITLE
[TM-47] - View list of applications

### PR DESCRIPTION
**Pull request by DannyGB**
Requires TM-47 LTM Api and Web branches

### CODE REVIEW

File: src/SFA.DAS.LevyTransferMatching.Api/Authentication/PolicyNames.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace SFA.DAS.LevyTransferMatching.Api.Authentication
+{
+    public static class PolicyNames
+    {
+        public const string PledgeAccount = "PledgeAccount";
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching.Api/Authentication/PolicyNames.cs:5]: I'd be tempted to rename the policy to something where the intent is more clearly expressed, maybe "ManagePledge" or something like that. Likewise the requirement. "PledgeAccount" is just a bit non-obvious in what it _does_ I think?

Other than that though, looks great.

File: src/SFA.DAS.LevyTransferMatching.Api/Controllers/PledgeController.cs [hunk 2]
```diff
@@ -162,5 +167,25 @@ public async Task<IActionResult> JobRole()
                 return new StatusCodeResult((int)HttpStatusCode.InternalServerError);
             }
         }
+
+        [Authorize(Policy = PolicyNames.PledgeAccount)]
+        [HttpGet]
+        [Route("accounts/{accountId}/pledges/{pledgeId}/applications")]
+        public async Task<IActionResult> PledgeApplications(int pledgeId)
+        {
+            var applications = await _mediator.Send(new GetApplicationsQuery { PledgeId = pledgeId });
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching.Api/Controllers/PledgeController.cs:176]: The pattern we've used elsewhere has avoided multiple calls to the mediator from the controller in favour of a single call which does everything required and returns a single result. This approach also exposes an inner api type (GetStandardsListItem) nested inside of an outer api response type.

File: src/SFA.DAS.LevyTransferMatching/Application/Queries/Pledges/GetApplications/GetApplicationsQueryHandler.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    {
+        private readonly ILevyTransferMatchingService _levyTransferMatchingService;
+
+        public GetApplicationsQueryHandler(ILevyTransferMatchingService levyTransferMatchingService)
+        {
+            _levyTransferMatchingService = levyTransferMatchingService;
+        }
+
+        public async Task<GetApplicationsQueryResult> Handle(GetApplicationsQuery request, CancellationToken cancellationToken)
+        {
```
Comment (chrisfoster76) [inline src/SFA.DAS.LevyTransferMatching/Application/Queries/Pledges/GetApplications/GetApplicationsQueryHandler.cs:19]: As per previous comment, we could go get the standards here as well (and await them both together) and then assemble a single response.

**Review by chrisfoster76 — CHANGES_REQUESTED** [review]

File: src/SFA.DAS.LevyTransferMatching.Api/Controllers/PledgeController.cs (hunk 2 above)
Comment (DannyGB) [reply src/SFA.DAS.LevyTransferMatching.Api/Controllers/PledgeController.cs:176]: So you would create a second object to hold the Standard data and convert from GetStandardsListItem to it prior to returning back from the query handler?

File: src/SFA.DAS.LevyTransferMatching/Application/Queries/Pledges/GetApplications/GetApplicationsQueryHandler.cs (hunk 3 above)
Comment (DannyGB) [reply src/SFA.DAS.LevyTransferMatching/Application/Queries/Pledges/GetApplications/GetApplicationsQueryHandler.cs:19]: Agreed, I'll implement this

**Review by chrisfoster76 — APPROVED** [review]

**Review by ConorTill — APPROVED** [review]

**Pull request merged**